Aqovia.Utilities.SagaMachine.Tests/StateMachineTests.cs

```diff
@@ -338,5 +356,38 @@ public void SagaMachineShouldHaveInitialiseStateId()
             _keyValueStoreMock.Verify(o => o.Remove(It.IsAny<string>(), It.IsAny<string>()), Times.Once);
         }
 
+        [Fact]
+        public void SagaMachine_Should_Not_Stop_Saga_If_An_Exception_Happens()
+        {
+            //Arrange
+            _keyValueStoreMock.Setup(o => o.GetValue<TestState>(It.IsAny<string>()))
+                .Returns(new HashedValue<TestState>
+                {
+                    Hash = Guid.Empty.ToString(),
+                    Value = new TestState()
+                });
+
+            // We need the below to set the return value
+            _keyValueStoreMock.Setup(kv => kv.Remove(It.IsAny<string>(), It.IsAny<string>())).Returns(true);
+
+            var messagePublisher = new MessagePublisherFake();
+
+            //Act
+            _sagaMachine = new SagaMachine<TestState>(_keyValueStoreMock.Object, messagePublisher.PublishMessage, _mockEventloggerFactory.Object);
+            _sagaMachine
+               .WithMessage<HelloMessage>((proccess, msg) => proccess
+                   .Publish((msgForPub, state) => new[] { new GoodbyeMessage() })
+                   .Publish((msgForPub, state) => new[] { new SomeOtherMessage() })
```

Comment:
You're right it does not make sense now.
I was the case initially when I was trying to simulate a production scenario with multiple "Publish" and different task completion duration.